### PR TITLE
WEBUI-200: add discernible text to buttons

### PR DESCRIPTION
**Pull request by semisse**

### CODE REVIEW

**Review by richardsd — APPROVED** [review]

**Comment by semisse** [comment]
I just pushed some new modifications that essentially are replacing the `aria-label` with `aria-labelledby`. This way we save another request to `i18n` and we keep our code more DRY.

File: addons/nuxeo-platform-3d/elements/nuxeo-3d-transmission-formats.js [hunk 1]
```diff
@@ -124,12 +124,26 @@ Polymer({
...
-                <paper-tooltip>[[i18n('threeDViewLayout.transmissionFormats.preview')]]</paper-tooltip>
+                <paper-icon-button
+                  icon="icons:visibility"
+                  on-tap="_loadFormat"
+                  noink
+                  aria-labelledby="previewTooltip"
+                ></paper-icon-button>
+                <paper-tooltip id="previewTooltip"
+                  >[[i18n('threeDViewLayout.transmissionFormats.preview')]]</paper-tooltip
+                >
```
Comment (mnixo) [inline addons/nuxeo-platform-3d/elements/nuxeo-3d-transmission-formats.js:135]:
```suggestion
                <paper-tooltip id="previewTooltip">
                  [[i18n('threeDViewLayout.transmissionFormats.preview')]]
                </paper-tooltip>
```
or
```suggestion
                <paper-tooltip
                  id="previewTooltip"
                >[[i18n('threeDViewLayout.transmissionFormats.preview')]]</paper-tooltip>
```

File: addons/nuxeo-platform-3d/elements/nuxeo-3d-transmission-formats.js [hunk 2]
```diff
@@ -124,12 +124,26 @@ Polymer({
...
-                <paper-tooltip>[[i18n('threeDViewLayout.transmissionFormats.download')]]</paper-tooltip>
+                <paper-icon-button
+                  icon="icons:file-download"
+                  on-tap="_downloadFormat"
+                  noink
+                  aria-labelledby="downloadTooltip"
+                ></paper-icon-button>
+                <paper-tooltip id="downloadTooltip"
+                  >[[i18n('threeDViewLayout.transmissionFormats.download')]]</paper-tooltip
+                >
```
Comment (mnixo) [inline addons/nuxeo-platform-3d/elements/nuxeo-3d-transmission-formats.js:146]: Same as above.

File: addons/nuxeo-template-rendering/elements/nuxeo-template-param-editor.js [hunk 3]
```diff
@@ -176,8 +176,9 @@ Polymer({
                 icon="icons:create"
                 on-tap="_editParam"
                 disabled="[[!_canEdit(param, params)]]"
+                aria-labelledby="editTooltip"
               ></paper-icon-button>
-              <paper-tooltip for="[[_computeBtnId(param, 'edit')]]"
+              <paper-tooltip for="[[_computeBtnId(param, 'edit')]]" id="editTooltip"
                 >[[i18n('templateRenderingPage.parameters.edit.tooltip')]]</paper-tooltip
               >
```
Comment (mnixo) [inline addons/nuxeo-template-rendering/elements/nuxeo-template-param-editor.js:183]: Same as above (although it was like this already).

File: addons/nuxeo-template-rendering/elements/nuxeo-template-param-editor.js [hunk 4]
```diff
@@ -186,17 +187,25 @@ Polymer({
                   icon="icons:delete"
                   on-tap="_deleteParam"
                   disabled="[[!_canEdit(param, params)]]"
+                  aria-labelledby="removeTooltip"
                 ></paper-icon-button>
-                <paper-tooltip for="[[_computeBtnId(param, 'remove')]]"
+                <paper-tooltip for="[[_computeBtnId(param, 'remove')]]" id="removeTooltip"
                   >[[i18n('templateRenderingPage.parameters.remove.tooltip')]]</paper-tooltip
                 >
```
Comment (mnixo) [inline addons/nuxeo-template-rendering/elements/nuxeo-template-param-editor.js:194]: Same as above (although it was like this already).

File: addons/nuxeo-template-rendering/elements/nuxeo-template-param-editor.js [hunk 5]
```diff
@@ -186,17 +187,25 @@ Polymer({
...
-          <paper-tooltip for="addParamBtn">[[i18n('templateRenderingPage.parameters.add.tooltip')]]</paper-tooltip>
+          <paper-icon-button
+            id="addParamBtn"
+            icon="icons:add-circle"
+            on-tap="_addParam"
+            aria-labelledby="addParametersTooltip"
+          ></paper-icon-button>
+          <paper-tooltip for="addParamBtn" id="addParametersTooltip"
+            >[[i18n('templateRenderingPage.parameters.add.tooltip')]]</paper-tooltip
+          >
```
Comment (mnixo) [inline addons/nuxeo-template-rendering/elements/nuxeo-template-param-editor.js:208]: Same as above.

File: elements/diff/nuxeo-diff.js [hunk 6]
```diff
@@ -171,8 +171,11 @@ Polymer({
               id="switchSidesButtonH"
               icon="nuxeo:switch-sides"
               on-tap="_switchSides"
+              aria-labelledby="switchSidesButtonHTooltip"
             ></paper-icon-button>
-            <nuxeo-tooltip for="switchSidesButtonH">[[i18n('diff.controls.switchSides.tooltip')]]</nuxeo-tooltip>
+            <nuxeo-tooltip for="switchSidesButtonH" id="switchSidesButtonHTooltip"
+              >[[i18n('diff.controls.switchSides.tooltip')]]</nuxeo-tooltip
+            >
```
Comment (mnixo) [inline elements/diff/nuxeo-diff.js:178]: Same as above.

File: elements/diff/nuxeo-diff.js [hunk 7]
```diff
@@ -187,8 +190,11 @@ Polymer({
             id="switchSidesButtonV"
             icon="nuxeo:switch-sides"
             on-tap="_switchSides"
+            aria-labelledby="switchSidesButtonVTooltip"
           ></paper-icon-button>
-          <nuxeo-tooltip for="switchSidesButtonV">[[i18n('diff.controls.switchSides.tooltip')]]</nuxeo-tooltip>
+          <nuxeo-tooltip for="switchSidesButtonV" id="switchSidesButtonVTooltip"
+            >[[i18n('diff.controls.switchSides.tooltip')]]</nuxeo-tooltip
+          >
```
Comment (mnixo) [inline elements/diff/nuxeo-diff.js:197]: Same as above.

File: elements/directory/nuxeo-vocabulary-management.js [hunk 8]
```diff
@@ -118,17 +118,19 @@ Polymer({
                       id="edit-button-[[index]]"
                       icon="nuxeo:edit"
                       on-tap="_editEntry"
+                      aria-labelledby="editButtonTooltip"
                     ></paper-icon-button>
-                    <nuxeo-tooltip for="edit-button-[[index]]"
+                    <nuxeo-tooltip for="edit-button-[[index]]" id="editButtonTooltip"
                       >[[i18n('vocabularyManagement.editEntry')]]</nuxeo-tooltip
                     >
```
Comment (mnixo) [inline elements/directory/nuxeo-vocabulary-management.js:125]: Same as above (although it was like this already).

File: elements/directory/nuxeo-vocabulary-management.js [hunk 9]
```diff
@@ -118,17 +118,19 @@ Polymer({
...
                       id="delete-button-[[index]]"
                       name="delete"
                       icon="nuxeo:delete"
                       on-tap="_deleteEntry"
+                      aria-labelledby="deleteButtonTooltip"
                     ></paper-icon-button>
-                    <nuxeo-tooltip for="delete-button-[[index]]"
+                    <nuxeo-tooltip for="delete-button-[[index]]" id="deleteButtonTooltip"
                       >[[i18n('vocabularyManagement.deleteEntry')]]</nuxeo-tooltip
                     >
```
Comment (mnixo) [inline elements/directory/nuxeo-vocabulary-management.js:135]: Same as above (although it was like this already).

File: elements/nuxeo-app/nuxeo-menu-icon.js [hunk 10]
```diff
@@ -66,8 +66,10 @@ Polymer({
     </style>
 
     <a href$="[[_href(urlFor, route, link)]]">
-      <paper-icon-button noink id="button" name$="[[name]]"></paper-icon-button>
-      <nuxeo-tooltip for="button" position="right" offset="0" animation-delay="0">[[i18n(label)]]</nuxeo-tooltip>
+      <paper-icon-button noink id="button" name$="[[name]]" aria-labelledby="tooltip"></paper-icon-button>
+      <nuxeo-tooltip for="button" position="right" offset="0" animation-delay="0" id="tooltip"
+        >[[i18n(label)]]</nuxeo-tooltip
+      >
```
Comment (mnixo) [inline elements/nuxeo-app/nuxeo-menu-icon.js:72]: Same as above.

File: elements/nuxeo-collections/nuxeo-collections.js [hunk 11]
```diff
@@ -178,9 +178,17 @@ Polymer({
...
+          icon="icons:arrow-back"
+          id="backToCollections"
+          on-tap="displayCollections"
+          aria-labelledby="backToCollectionsTooltip"
+        >
         </paper-icon-button>
-        <nuxeo-tooltip for="backToCollections">[[i18n('collections.backToCollections')]]</nuxeo-tooltip>
+        <nuxeo-tooltip for="backToCollections" id="backToCollectionsTooltip"
+          >[[i18n('collections.backToCollections')]]</nuxeo-tooltip
+        >
```
Comment (mnixo) [inline elements/nuxeo-collections/nuxeo-collections.js:191]: Same as above.

File: elements/nuxeo-document-create-button/nuxeo-document-create-button.js [hunk 12]
```diff
@@ -81,9 +81,17 @@ Polymer({
...
+        noink
+        icon="nuxeo:add"
+        on-tap="_displayWizard"
+        aria-labelledby="createBtnTooltip"
+      ></paper-fab>
       <!-- nuxeo-tooltip does not play nice (in shadycss) when attached to elements that are position: absolute -->
-      <paper-tooltip for="createBtn" position="left">[[i18n('documentCreateButton.tooltip')]]</paper-tooltip>
+      <paper-tooltip for="createBtn" position="left" id="createBtnTooltip"
+        >[[i18n('documentCreateButton.tooltip')]]</paper-tooltip
+      >
```
Comment (mnixo) [inline elements/nuxeo-document-create-button/nuxeo-document-create-button.js:94]: Same as above.

File: elements/nuxeo-note-editor/nuxeo-note-editor.js [hunk 13]
```diff
@@ -93,8 +93,11 @@ Polymer({
             icon="[[_computeHtmlEditIcon(_viewMode)]]"
             on-tap="_toggleHtmlSource"
             hidden$="[[!_canEdit(document)]]"
+            aria-labelledby="editHtmlTooltip"
           ></paper-icon-button>
-          <paper-tooltip for="editHtml" position="right">[[_computeHtmlEditLabel(_viewMode, i18n)]]</paper-tooltip>
+          <paper-tooltip for="editHtml" position="right" id="editHtmlTooltip"
+            >[[_computeHtmlEditLabel(_viewMode, i18n)]]</paper-tooltip
+          >
```
Comment (mnixo) [inline elements/nuxeo-note-editor/nuxeo-note-editor.js:100]: Same as above.

File: elements/search/nuxeo-search-form.js [hunk 14]
```diff
@@ -271,19 +271,30 @@ Polymer({
...
+              icon="nuxeo:filter"
+              id="toogleFilter"
+              on-tap="_displayFiltersTapped"
+              aria-labelledby="toogleFilterTooltip"
+            >
             </paper-icon-button>
-            <nuxeo-tooltip for="toogleFilter">[[i18n('searchForm.displayFilterView')]]</nuxeo-tooltip>
+            <nuxeo-tooltip for="toogleFilter" id="toogleFilterTooltip"
+              >[[i18n('searchForm.displayFilterView')]]</nuxeo-tooltip
+            >
```
Comment (mnixo) [inline elements/search/nuxeo-search-form.js:284]: Same as above.

**Review by mnixo — COMMENTED** [review]
The HTML formatting just seems a bit funky here and there.
EDIT: The `prettier` formatting is working as intended.

**Review by mnixo — APPROVED** [review]

File: i18n/messages.json [hunk 15]
```diff
@@ -281,16 +281,21 @@
   "command.cancel": "Cancel",
   "command.clear": "Clear",
   "command.close": "Close",
+  "command.confirm": "Confirm",
```
Comment (Gabez0r) [inline i18n/messages.json:284]: We need to ping @manonlumeau once this is merged.

**Review by Gabez0r — APPROVED** [review]

**Review by richardsd — APPROVED** [review]

**Comment by semisse** [comment]
https://qa.nuxeo.org/jenkins/job/web-ui-pipeline-sfialho/246/ 🔵

**Pull request merged**